### DISCUSSION TITLE
how to remove "bookmark this page" icon from the location bar

### DISCUSSION THREAD

**Discussion by joefiesta** [Aris-t2/CustomCSSforFx#677] [Q&A] [answered]
How, if possible, can I remove the "bookmark this page" icon (big star on the right) from the location bar?

**Comment by Aris-t2** [comment]
Use this code:

```css
#star-button-box {
  display :none !important;
}
```

**Reply by joefiesta** [reply]
that does not work.   I think a #urlbar parameter is needed, but not sure where.

I tried:

> #urlbar #star-button-box {
>   display :none !important;
> }
> 

that does not work as well.

**Reply by Aris-t2** [reply]
It works the way it should here:
![image](https://github.com/user-attachments/assets/aadbc20a-f9e8-4a98-a3b9-2b2151c70496)
![image](https://github.com/user-attachments/assets/8666c338-8ab7-45c6-91bd-fd7fcac6ea42)

**Reply by joefiesta** [reply]
Well, something I do is interfering then.  Attached are my .CSS files.   Note:  I DO NOT MODIFY ANY ARIS .CSS FILES other than  UserChrome.css.   And that is only to add  my @import
...
well, I can't attach my .CSS files.  Github won't let me.  Am I supposed to copy them to another filetype, like .TXT?   That seems absurd.  Pasting them seems infantile.

**Reply by joefiesta** [reply]
I think I figured out how to zip the  files.   

[JSP_userChrome.zip](https://github.com/user-attachments/files/17016413/JSP_userChrome.zip)

**Reply by Aris-t2** [reply]
Just do a tiny test.
Backup your `chrome` folder and remove everything from there, extract this projects full package, add the code to the bottom of `userChrome.css` or to new file `my_userChrome.css`.

**Reply by joefiesta** [reply]
I tested as follows.

===========================
first test
/*  The purpose of creating this UserChrome is to keep the Aris_userChrome.css                  */
/*  file completely unmodified.                                                                 */
/*                                                                                              */
@import "./Aris_userChrome_4.5.2.css";
/*                                                                                              */
/* Remove BOOKMARK THIS PAGE icon from location bar.  added 20240915 */
#star-button-box {
  display :none !important;
}
@import "./JSP_userChrome.css";

with this userchrome.css, the button was gone.  HOWEVER, none of my code in JSP_userChrome.css was executed.

======================================
test 2

/*  The purpose of creating this UserChrome is to keep the Aris_userChrome.css                  */
/*  file completely unmodified.                                                                 */
/*                                                                                              */
@import "./Aris_userChrome_4.5.2.css";
/*                                                                                              */
/* Remove BOOKMARK THIS PAGE icon from location bar.  added 20240915 */
#star-button-box {
  display :none !important;
}

this userChome.css removed the button.   But, of course, again my code (which isn't here imported) did not execute. 
==========================
note:  i did not  upgrade.  I used 4.5.2.   As my code says, I DO NOT TOUCH ANY ARIS FILES except userchrome.css.

so,
a.   why did my coded not even get execute in test 1?
b.  it appears something in my code is in conflict with the suggested addition.  I sent, in  a zip file, all my code.  

thx
Joe P

**Reply by joefiesta** [reply]
I tried another test....
using my original setup, I added 
 
   #urlbar #star-button-box {
   display :none !important;
   }

to the very top of my imported file.   This time, it removed the button, but again, NONE OF MY CODE GOT PROCESSED.

**Reply by Aris-t2** [reply]
I could find a few errors inside the `css` files in your zip file.

`erChrome.css`
Error in line 27. `* /` has to be `*/`.

`JSP.custom_tab_color_settings.css`
There are too many `;` within the `:root` area.

`JSP_buttons_on_navbar_classic_appearance.css`
Error in line 19. There are two `*/` at the end.

I suggest do start over and try to copy and paste or to import one file by one. Your `erChrome.css` import far more stuff than the zip file offers.

**Reply by joefiesta** [reply]
thanks.... fixed that stuff, but still don't get the STAR to go away.

1.  How did you find my errors?    (manually or programmatically?)
2.  I found numerous
      `#urlbar #star-button:`
statements in C:Users\Joey\AppData\Roaming\Mozilla\Firefox\Profiles\7agyzsl4.Joey 10\Chrome\css\locationbar\icons_colorized.css

do they relate to `#star-button-box {
  display :none !important;
} `?

**Reply by Aris-t2** [reply]
Use an editor with code highlighting like Notepad++.
I looked through your code "manually".

**Reply by joefiesta** [reply]
I found the solution.  

1.  The code provided was, as I never doubted, correct.   However, it became a matter of  WHERE I PLACED IT.  
2.   It also turned out that my editorial CSS error were not relevant--but I did, of course--fix them!  Thx ARIS!!!
3.   I had never realized that I had commented out CSS lines incorrectly.   that happens when you're 73 and have used Rexx and Autohotkey forever.    I thought--as in AHK--a leading semicolon did the trick!
4.   I don't understand why where I placed the code mattered.    If anyone would like to explain that, I will be happy to send whatever code I have.  I think the easiest would be zipping my entire Chrome folder.  Bits and pieces make analysis cumbersome.

**Reply by joefiesta** [reply]
So, I appeared solved.  Well, it is not.

Removing the Star-button-box now means bookmarking a tab is problematic.
I can bookmark a tab (using the Bookmarks "button" in the FILE menu.  (file, edit, view...etc).    But
1.  I can not choose where to place the bookmark
2.  After it is created I can not edit the bookmark.  
the popups to do so just go away instantly (or don't appear at all).

**Reply by joefiesta** [reply]
does anyone have a response about my last entry?

**Reply by Aris-t2** [reply]
If you remove the button, you have to use alternative ways to do what you intend to do, but keeping it visible might be the better solution for you.

1. Ctrl+D, if the bookmarked page is currently open, can still open the menu. You also can manually drag and drop existing bookmarks between bookmark areas, folders etc.
2. Ctrl+D, if the bookmarked page is currently open, or right-click a bookmark and edit it.

**Reply by joefiesta** [reply]
@Aris-t2  - Only partly true here.  
   1.  If page is not bookmarked, CTRL + D bookmarks it.  However, the menu simply flashes and disappears.  I have no choice of WHERE to bookmark it (and that is important to me).  I subsequently have difficulty finding the bookmark.
  2.  If the page IS bookmarked,  CTRL + D opens the menu, but choosing "Edit this bookmark" again makes the next menu simply flash and disappear.  And CTRL + D in this case offers no option to DELETE the bookmark (or is that usually in the edit menu?).

**Reply by Aris-t2** [reply, marked as answer]
The flashing and disappearing does not happen in my tests, but we can try a different approach.

Using this code instead of `display: none` will keep the button, but hide it and reduce its used space to a minimum. That way all CTRL+D functionality should stay. I still think keeping the button is the best solution to keep usability.

```css
#star-button-box {
  width: 0.1px !important;
  margin: 0 !important;
  padding: 0 !important;
  opacity: 0 !important;
} 
```

**Reply by joefiesta** [reply]
Thanks.... that works.    the button is still there, but so small!!  ha ha  Clever!
I thought at first it did not work.  My testing uncovered what is probably a flaw with Firefox. 

If you open Bookmarks/ Bookmark current tab...  and then, doing nothing on the popup,  move the cursor OUTSIDE  of the popup, the SAVE option is automatically processed.  (this is from the FILE menu... I presume the button does this also.)
